js/models/messages.js

```diff
@@ -1031,65 +1031,52 @@
-
+      const chatMessage = new libsession.Messages.Outgoing.ChatMessage({
+        body,
+        timestamp: this.get('sent_at'),
+        attachments,
+        quote: quoteWithData,
+        preview: previewWithData,
+        expireTimer: this.get('expireTimer'),
+      });
       // Special-case the self-send case - we send only a sync message
       if (recipients.length === 1 && recipients[0] === this.OUR_NUMBER) {
-        const [number] = recipients;
-        const dataMessage = await textsecure.messaging.getMessageProto(
-          number,
-          body,
-          attachments,
-          quoteWithData,
-          previewWithData,
-          this.get('sent_at'),
-          this.get('expireTimer'),
-          profileKey
-        );
-        return this.sendSyncMessageOnly(dataMessage);
+        this.trigger('pending');
+        // FIXME audric add back profileKey
+        await this.markMessageSyncOnly();
+        const syncMessage = libsession.Utils.SyncMessageUtils.from(chatMessage);
+        return libsession.getMessageQueue().sendSyncMessage(syncMessage);
       }
 
-      let promise;
-      const options = conversation.getSendOptions();
-      options.messageType = this.get('type');
-
       if (conversation.isPrivate()) {
         const [number] = recipients;
-        promise = textsecure.messaging.sendMessageToNumber(
-          number,
-          body,
-          attachments,
-          quoteWithData,
-          previewWithData,
-          this.get('sent_at'),
-          this.get('expireTimer'),
-          profileKey,
-          options
-        );
-      } else {
-        // Because this is a partial group send, we manually construct the request like
-        //   sendMessageToGroup does.
-
-        promise = textsecure.messaging.sendMessage(
-          {
-            recipients,
-            body,
-            timestamp: this.get('sent_at'),
-            attachments,
-            quote: quoteWithData,
-            preview: previewWithData,
-            needsSync: !this.get('synced'),
-            expireTimer: this.get('expireTimer'),
-            profileKey,
-            group: {
-              id: this.get('conversationId'),
-              type: textsecure.protobuf.GroupContext.Type.DELIVER,
-            },
-          },
-          options
-        );
+        const recipientPubKey = new libsession.Types.PubKey(number);
+        this.trigger('pending');
+
+        return libsession
+          .getMessageQueue()
+          .sendUsingMultiDevice(recipientPubKey, chatMessage);
       }
 
-      return this.send(conversation.wrapSend(promise));
+      this.trigger('pending');
+      // TODO should we handle open groups message here too? and mediumgroups
+      // Not sure there is the concept of retrySend for those
+      const closedGroupChatMessage = new libsession.Messages.Outgoing.ClosedGroupChatMessage(
+        {
+          chatMessage,
+          groupId: this.get('conversationId'),
```

Comment:
are you sure `groupId` is `conversationId`? i thought `conversationId` had a prefix attached to it or something 🤔

Reply:
It's already what was used before: https://github.com/loki-project/session-desktop/pull/1197/files/1650d758a54bf27236ab4570defd9ceefdbb6871#diff-fdb432d7888e10d18d02c7becd34c5e4L1084